src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/RealWorldContextTests.cs

```diff
@@ -838,6 +838,44 @@ void RunTest(ClassWithNullableProperties expected)
             }
         }
 
+#if NET6_0_OR_GREATER
+        [Fact]
+        public virtual void ClassWithDateOnlyAndTimeOnlyValues_Roundtrip()
+        {
+            RunTest(new ClassWithDateOnlyAndTimeOnlyValues
+            {
+                DateOnly = DateOnly.Parse("2022-05-10"),
```

Comment:
>Parse [](http://example.com/codeflow?start=36&length=5)

why we need to parse values here we are going to use? Also would it make sense to use dynamic values? Something like:

```C#
DateOnly = DateOnly.FromDateTime(DateTime.UtcNow);
TimeOnly = TimeOnly.FromDateTime(DateTime.UtcNow);
...
```

Reply:
Intentionally avoiding the use of "Now" dates to make test runs deterministic.

Reply:
why?

Reply:
A test might contain bugs when dealing with boundary values (e.g. during daylight saving changes, leap days, leap seconds, new year's) and we want to avoid them manifesting as flaky CI.

Reply:
Daylight saving has nothing to do with `DateOnly/TimeOnly`. I cannot think in any way using `UtcNow` will cause any problem in such tests.

Reply:
Maybe, but as a general rule we should avoid using `Now` in unit tests, in the same way we should avoid using `new Random()` instances without a `seed` parameter. Adding nondeterminism always introduces the potential of breakage in unpredictable ways.

Reply:
I'll leave it to you to decide. I see using UtcNow in tests like that is adding more value to ensure we are working correctly with different values and not only one hardcoded value. Again, I am not seeing any harm using UtcNow but it is your call.

Reply:
We tend to use hardcoded values in STJ so I'll leave it as-is for now. There's definitely value in stress testing a library with randomized inputs, but for the benefit of having predictable CI runs these need to be pseudo-random. Testing frameworks like FsCheck are capable of doing this, and this is already being done in other libraries like [System.Formats.Cbor](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Formats.Cbor/tests/PropertyTests/CborPropertyTests.cs)